amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseCoordinator.java

```diff
@@ -76,6 +83,8 @@ public class DynamoDBLeaseCoordinator implements LeaseCoordinator {
     private long initialLeaseTableReadCapacity;
     private long initialLeaseTableWriteCapacity;
     protected final MetricsFactory metricsFactory;
+    @Nullable
+    private final StreamProcessingMode streamProcessingMode;
```

Comment:
I strong feel this should be done in scheduler and not here, lets discuss more

Reply:
Sure, let's talk over a call.

Reply:
Have moved the replacement logic to scheduler.